core/include/kernel/delay.h

```diff
@@ -33,6 +33,28 @@
 #include <kernel/delay_arch.h>
 #include <stdbool.h>
 #include <stdint.h>
+#include <util.h>
+
+#ifdef CFG_CORE_HAS_GENERIC_TIMER
+/* Convert microsecond base delay @us into architecture time tick counts */
+static inline uint64_t delay_us2cnt(uint32_t us)
+{
+	return ((uint64_t)us * (uint64_t)delay_cnt_freq()) / ULL(1000000);
+}
+
+/* Return delay tick counter for a timeout expiration in @us microseconds */
+static inline uint64_t timeout_init_us(uint32_t us)
+{
+	return barrier_read_counter_timer() + delay_us2cnt(us);
```

Comment:
`delay_cnt_read()` instead of `barrier_read_counter_timer()`?

Reply:
Rigth, my mistake. I'll update.